src/Security/TokenChecker.php

```diff
@@ -97,6 +115,28 @@ private function getToken(string $sessionKey): ?TokenInterface
             return null;
         }
 
+        if ($this->trustResolveer->isAnonymous($token)) {
+            return null;
+        }
+
         return $token;
     }
+
+    /**
+     * Gets the username of a token in the session.
+     *
+     * @param string $userClass
+     *
+     * @return string|null
+     */
+    private function getUsername(string $userClass): ?string
+    {
+        $token = $this->getToken(\constant($userClass.'::SECURITY_SESSION_KEY'));
```

Comment:
This implies that `$userClass` always has a `SECURITY_SESSION_KEY` constant. It would be better (thinking of unit tests) to pass the session key as second argument.

Reply:
I have added a check in be0c6d4de2cafccf7bef134d31f25e8c20f4f0ae. I don't think a second argument would be good, as you could then misconfigure it (mix a `FrontendUser` class with a backend user security session key).